pkg/controller/certificates/rootcacertpublisher/root_ca_cert_publisher.go

```diff
@@ -175,45 +160,50 @@ func (c *Publisher) processNextWorkItem() bool {
 	}
 	defer c.queue.Done(key)
 
-	err := c.syncHandler(key.(string))
-	if err == nil {
-		c.queue.Forget(key)
+	if err := c.syncHandler(key.(string)); err == nil {
+		utilruntime.HandleError(fmt.Errorf("syncing %q failed: %v", key, err))
+		c.queue.AddRateLimited(key)
 		return true
 	}
 
-	utilruntime.HandleError(fmt.Errorf("%v failed with : %v", key, err))
-	c.queue.AddRateLimited(key)
-
+	c.queue.Forget(key)
 	return true
 }
 
-func (c *Publisher) syncNamespace(key string) error {
+func (c *Publisher) syncNamespace(ns string) error {
 	startTime := time.Now()
 	defer func() {
-		glog.V(4).Infof("Finished syncing namespace %q (%v)", key, time.Since(startTime))
+		glog.V(4).Infof("Finished syncing namespace %q (%v)", ns, time.Since(startTime))
 	}()
 
-	ns, err := c.nsLister.Get(key)
```

Comment:
It seems we don't need the nsLister and nsListerSynced in Publisher now

Reply:
Fixed.